src/agent/onefuzz-supervisor/src/worker.rs

```diff
@@ -228,25 +232,97 @@ impl IWorkerRunner for WorkerRunner {
         cmd.arg("managed");
         cmd.arg("config.json");
         cmd.arg(setup_dir);
+        cmd.stderr(Stdio::piped());
+        cmd.stdout(Stdio::piped());
 
         let child = cmd.spawn().context("onefuzz-agent failed to start")?;
-        let child = Box::new(child);
+        Ok(Box::new(RedirectedChild::new(child)?))
+    }
+}
+
+/// Child process with redirected output streams, tailed by two worker threads.
+struct RedirectedChild {
+    /// The child process.
+    child: Child,
+
+    /// Worker threads which continuously read from the redirected streams.
+    streams: Option<StreamReaderThreads>,
+}
 
-        Ok(child)
+impl RedirectedChild {
```

Comment:
This would be *really* good to write a unit test around, something akin to:
`perl -e 'print STDOUT "a"x4090 . "b"x4090; print STDERR "c"x4090 . "d"x4090'`

Reply:
Agreed, this and the circular buffer both need unit tests (and I need to end-to-end test it still).